hfs/preprocessing.py

```diff
@@ -123,28 +123,44 @@ def get_hierarchy(self):
             In this case the hierarchy graph has not been updated yet.
         """
         if self.is_fitted_:
-            output_hierarchy = self._hierarchy
+            output_hierarchy = self._hierarchy_graph
             output_hierarchy.remove_node("ROOT")
-            return nx.to_numpy_array(self._hierarchy)
+            return nx.to_numpy_array(self._hierarchy_graph)
         else:
             raise RuntimeError("Instance has not been fitted.")
 
     def _extend_dag(self):
         """Adds missing nodes to the hierarchy graph.
 
-        For columns that no have a corresponding node in the hierarchy a
+        For columns that don't have a corresponding node in the hierarchy a
         node is added right under the "ROOT" node.
+        We then update the columns mapping to include the new nodes.
+        If a node in the hierarchy has a name conflict with a column in the
+        dataset we add a node with the next available id.
         """
-        max = len(self._hierarchy.nodes) - 1
-        for x in range(len(self._columns)):
-            if self._columns[x] == -1:
-                if x in self._hierarchy.nodes:
-                    self._hierarchy.add_edge("ROOT", max)
-                    self._columns[x] = max
-                    max += 1
+        # -1 to account for the "ROOT" node
```

Comment:
Change suggestion for line 141:

Was:
`# -1 to account for the "ROOT" node`

To sth. like this:
`# The list of nodes of length N contains the "ROOT" node and all other N-1 nodes with indexes (or: names?) ranging from 0 to N-2, thus we add '-1' to identify the next available node index.`

Reply:
Done!